### PR TITLE
NO-ISSUE: Add a dev-webapp to DMN Dev Deployments Form webapp

### PR DESCRIPTION
**Pull request by ljmotta**
## On this PR
Added a showcase app with hot reload to `packages/dmn-dev-deployments-form-webapp` in the `pnpm start` script.

It will spawn two processes:
1. A Webpack dev server on `http://localhost:9008`
2. A Quarkus dev server on `http://localhost:9009`

Accessing the `http://localhost:9008`  and selecting the example will render the respective form which can be used.

This PR also improves code reusability as it extracts some common logic to `dmn-runner` package.

## Video
https://github.com/kiegroup/kie-tools/assets/24302289/464541ca-1c4e-4236-bb85-8f1e855662be

### CODE REVIEW

File: packages/dmn-dev-deployment-form-webapp/showcase/start.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+const path = require("path");
+const { env } = require("../env");
+
+const buildEnv = env;
+
+const mvn = spawn("mvn", [
+  "-f",
+  path.join(__dirname, "./quarkus-app"),
+  "clean",
+  "package",
```
Comment (tiagobento) [inline packages/dmn-dev-deployment-form-webapp/showcase/start.js]: Do we really need to go through the `package` phase? Isn't `clean quarkus:dev` enough?

File: packages/dmn-dev-deployment-form-webapp/src/DmnDevDeploymentRuntimeApi.tsx [hunk 2]
```diff
@@ -23,7 +23,8 @@ export interface FetchDmnResultArgs {
 }
 
 export async function fetchDmnResult(args: FetchDmnResultArgs): Promise<DecisionResult[]> {
-  const response = await fetch(routes.dmnResult.path({ modelName: args.modelName }), {
+  const a = routes.dmnResult.path({ modelName: args.modelName });
```
Comment (tiagobento) [inline packages/dmn-dev-deployment-form-webapp/src/DmnDevDeploymentRuntimeApi.tsx]: Maybe inline this variable?

File: packages/dmn-dev-deployment-form-webapp/showcase/start.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+const webpack = spawn("npx", [
+  "webpack",
+  "serve",
+  "-c",
+  path.join(__dirname, "./webapp/webpack.config.js"),
+  "--host",
+  "0.0.0.0",
+  "--env",
+  "dev",
```
Comment (tiagobento) [inline packages/dmn-dev-deployment-form-webapp/showcase/start.js]: We need a way to make sure that `pnpm start --env live` works here too. Probably appending argv to this list wouldn't be too bad. WDYT?

File: packages/dmn-dev-deployment-form-webapp/showcase/quarkus-app/pom.xml [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.kie.kogito</groupId>
+  <artifactId>dmn-dev-deployment-quarkus-app</artifactId>
+  <version>${revision}</version>
```
Comment (tiagobento) [inline packages/dmn-dev-deployment-form-webapp/showcase/quarkus-app/pom.xml]: Is this getting populated? I mean, usually there's an "install" script that will create a `.mvn` folder on each package, containing -Drevision=0.0.0 for development... Can you please check? I mean, this is almost irrelevant, but I didn't see anything on this PR that would address this property in the POM. This is more a question than a change request :)

**Review by tiagobento — CHANGES_REQUESTED** [review]
Nice job! Thanks for putting all this together.

May I ask you to rename "showcase" to "dev-webapp"? I know there are some other packages using "showcase", but I think it doesn't convey the right message. "Showcase" I think links to examples of 3rd party usage, but this is intended to be used to develop, not only to... showcase :)

File: packages/dmn-dev-deployment-form-webapp/src/DmnDevDeploymentRuntimeApi.tsx (hunk 2 above)
Comment (ljmotta) [reply packages/dmn-dev-deployment-form-webapp/src/DmnDevDeploymentRuntimeApi.tsx]: Oops. I extract the variable to check a few things.

**Review by tiagobento — APPROVED** [review]

**Pull request merged**